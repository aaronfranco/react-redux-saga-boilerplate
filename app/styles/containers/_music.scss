.app__music {
   margin-top:100px;
  .audioControls{
       text-align: center;
 }
  .card {
       width:100%;
       max-width:500px;
       box-shadow:0px 2px 5px 0px rgba(0,0,0,0.25);
       .card-header {
            padding:20px;
            font-family: "Roboto", "sans";
            font-size:32px;
            font-weight:bold;
       }
  }

}
