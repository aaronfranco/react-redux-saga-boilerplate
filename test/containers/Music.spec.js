import React from 'react';
import { shallow } from 'enzyme';

import { Music } from 'containers/Music';

const mockDispatch = jest.fn();

const props = {
  dispatch: mockDispatch,
  jane: {
    username: '',
    name: '',
    music: []
  },
};

function setup(ownProps = props) {
  return shallow(
    <Music {...ownProps} />,
    { attachTo: document.getElementById('react') }
  );
}

describe('Music', () => {
  const wrapper = setup();

  it('should be a Component', () => {
    expect(wrapper.instance() instanceof React.Component).toBe(true);
  });

  it('should render properly', () => {
    expect(wrapper.find('.app__music')).toBePresent();
  });

  // it('should render a Loader without data', () => {
  //   expect(wrapper.find('Loader')).toBePresent();
  // });

  it('should have dispatched an action on mount', () => {
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      payload: {},
      type: 'GET_JANE_REQUEST',
    });
  });

  it('should render some items when data arrives', () => {
    wrapper.setProps({
      jane: {
           username: "jane",
           name: "Jane Doe",
           music: [
                {
                     "title":"Track One",
                     "file":'414365__erokia__ambient-wave-28.wav'
                },
                {
                     "title":"Track Two",
                     "file":"414497__toiletrolltube__hellocatfood-b6b7-o.wav"
                },
                {
                     'title':"Track Three",
                     "file":'414514__eardeer__daybreak.wav'
                }
           ]
      },
    });

    expect(wrapper.find('.app__music__grid')).toMatchSnapshot();
  });


});
