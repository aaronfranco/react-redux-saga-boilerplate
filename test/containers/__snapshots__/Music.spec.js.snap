// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music should render some items when data arrives 1`] = `null`;
